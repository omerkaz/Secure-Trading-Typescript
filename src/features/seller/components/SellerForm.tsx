import { Field, Form, Formik } from 'formik';
import { Dispatch, SetStateAction } from 'react';
import { useDispatch } from 'react-redux';
import * as Yup from 'yup';

import Button from '@components/button/Button';

import {
    generateReferenceCode,
    generateSmsCode,
} from '@utils/codeGenerator/generateCode';
import { sellerFormInitialValues as initialValues } from '@utils/formsInitialValue/initialValues';
import { sellerValidationSchema } from '@utils/yup/sellerValidationSchema';

import FormElement from '../../../components/form/FormElement';
import { createSellerApplication } from '../sellerApplicationSlice';
import StatementText from './StatementText';

interface Props {
    nextForm: Dispatch<SetStateAction<number>>;
}

type Seller = Yup.InferType<typeof sellerValidationSchema>;

function SellerForm({ nextForm }: Props) {
    const dispatch = useDispatch();
    return (
        <Formik
            initialValues={initialValues}
            onSubmit={(values: Seller) => {
                // create sms and reference code and dispatch value, sms and reference code to seller application's  object
                dispatch(
                    createSellerApplication({
                        values,
                        SMSCode: generateSmsCode(),
                        referenceCode: generateReferenceCode(),
                    }),
                );
                // increase state for going to sms confirmation
                nextForm((prevValue) => prevValue + 1);
            }}
            validationSchema={sellerValidationSchema}
        >
            {({ values, errors, handleChange, touched }) => (
                <Form>
                    <div className="border rounded px-3 pt-2">
                        <div className="row mb-md-2 mb-sm-1">
                            <div className="col-md-6 col-sm-12 mt-2">
                                <div className="form-outline">
                                    <Field
                                        className="w-100 border rounded"
                                        as="select"
                                        name="legalOrRealPerson"
                                        values={values.legalOrRealPerson}
                                        style={{ height: '3.5vmin' }}
                                    >
                                        <option value="gerçek">
                                            Gerçek Kişi
                                        </option>
                                        <option value="tüzel">
                                            Tüzel Kişi
                                        </option>
                                    </Field>
                                </div>
                            </div>
                            <FormElement
                                as={'input'}
                                placeholder={'Tc Kimlik Numarası'}
                                name={'TCNumber'}
                                values={values.TCNumber}
                                errors={errors.TCNumber}
                                touched={touched.TCNumber}
                                handleChange={handleChange}
                            />
                        </div>
                        <div className="row mb-md-3 mb-sm-1">
                            <FormElement
                                as={'input'}
                                placeholder={'İsim Soyisim'}
                                name={'fullName'}
                                values={values.fullName}
                                errors={errors.fullName}
                                touched={touched.fullName}
                                handleChange={handleChange}
                            />
                            <FormElement
                                as={'input'}
                                placeholder={'Satıcı Telefon Numarası'}
                                name={'sellerPhoneNumber'}
                                values={values.sellerPhoneNumber}
                                errors={errors.sellerPhoneNumber}
                                touched={touched.sellerPhoneNumber}
                                handleChange={handleChange}
                            />
                        </div>
                        <div className="row mb-md-3 mb-sm-1">
                            <FormElement
                                as={'input'}
                                placeholder={'Satıcı Email'}
                                name={'sellerEmail'}
                                values={values.sellerEmail}
                                errors={errors.sellerEmail}
                                touched={touched.sellerEmail}
                                handleChange={handleChange}
                            />

                            <FormElement
                                as={'input'}
                                placeholder={'Alıcı Telefon Numarası'}
                                name={'buyerPhoneNumber'}
                                values={values.buyerPhoneNumber}
                                errors={errors.buyerPhoneNumber}
                                touched={touched.buyerPhoneNumber}
                                handleChange={handleChange}
                            />
                        </div>
                        <div className="row mb-md-3 mb-sm-1">
                            <FormElement
                                as={'input'}
                                placeholder={'Alıcı Email'}
                                name={'buyerEmail'}
                                values={values.buyerEmail}
                                errors={errors.buyerEmail}
                                touched={touched.buyerEmail}
                                handleChange={handleChange}
                            />
                            <FormElement
                                as={'input'}
                                placeholder={'Satıcı IBAN'}
                                name={'sellerIBAN'}
                                values={values.sellerIBAN}
                                errors={errors.sellerIBAN}
                                touched={touched.sellerIBAN}
                                handleChange={handleChange}
                            />
                        </div>
                        <div className="row mb-md-3 mb-sm-1">
                            <FormElement
                                as={'input'}
                                placeholder={'Araç Plaka Numarası'}
                                name={'vehicleNumber'}
                                values={values.vehicleNumber}
                                errors={errors.vehicleNumber}
                                touched={touched.vehicleNumber}
                                handleChange={handleChange}
                            />
                            <FormElement
                                as={'input'}
                                placeholder={'Satış Tutarı'}
                                name={'price'}
                                values={values.price}
                                errors={errors.price}
                                touched={touched.price}
                                handleChange={handleChange}
                            />
                        </div>
                    </div>
                    <div className="border rounded px-3 pt-2">
                        <StatementText
                            name={['consentTextKVKK', 'consentTextCompany']}
                            values={values}
                            errors={errors}
                            touched={touched}
                            handleChange={handleChange}
                        />
                    </div>
                    <div className="d-flex justify-content-center mt-2">
                        <Button children={'Gönder'}></Button>
                    </div>
                </Form>
            )}
        </Formik>
    );
}

export default SellerForm;
