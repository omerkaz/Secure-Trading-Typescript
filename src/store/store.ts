import { configureStore } from '@reduxjs/toolkit';
import sellerApplicationSlice from '@features/seller/sellerApplicationSlice';
import buyerApplicationSlice from '@features/buyer/buyerApplicationSlice';
import applicationStatusSlice from '@features/status/applicationStatusSlice';
import applicationCancelSlice from '@features/cancel/applicationCancelSlice';
const store = configureStore({
    reducer: {
        sellers: sellerApplicationSlice,
        buyers: buyerApplicationSlice,
        status: applicationStatusSlice,
        cancel:applicationCancelSlice,
    },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch


export default store;
