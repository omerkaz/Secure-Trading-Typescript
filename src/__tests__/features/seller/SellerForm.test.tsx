import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from 'store/store';

import SellerForm from '@features/seller/components/SellerForm';

describe('SellerForm elements renders correctly', () => {
    test('snapshot test', () => {
        const nextForm = jest.fn();
        const componentTree = render(
            <Provider store={store}>
                <SellerForm nextForm={nextForm} />
            </Provider>,
        );
        expect(componentTree).toMatchSnapshot();
    });

    test('elements test', () => {
        const nextForm = jest.fn();
        render(
            <Provider store={store}>
                <SellerForm nextForm={nextForm} />
            </Provider>,
        );
        const inputElement = screen.getByText('Tc Kimlik Numarası');
        expect(inputElement).toBeInTheDocument();
        const KVKKText = screen.getByText("Tarafıma hizmet sunulması amacıyla, bulut tabanlı yazılımların yurt dışında tutulması nedeniyle kişisel verilerimin yurt dışına aktarılmasına ilişkin Açık Rıza Metni'ni okudum. Kişisel verilerimin bu kapsamda aktarılmasına onay veriyorum.");
        expect(KVKKText).toBeInTheDocument();
    });
});
