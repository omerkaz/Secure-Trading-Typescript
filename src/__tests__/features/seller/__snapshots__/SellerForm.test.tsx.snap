// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellerForm elements renders correctly snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        action="#"
      >
        <div
          class="border rounded px-3 pt-2"
        >
          <div
            class="row mb-md-2 mb-sm-1"
          >
            <div
              class="col-md-6 col-sm-12 mt-2"
            >
              <div
                class="form-outline"
              >
                <select
                  class="w-100 border rounded"
                  name="legalOrRealPerson"
                  values="gerçek"
                >
                  <option
                    value="gerçek"
                  >
                    Gerçek Kişi
                  </option>
                  <option
                    value="tüzel"
                  >
                    Tüzel Kişi
                  </option>
                </select>
              </div>
            </div>
            <div
              class="col-md-6 col-sm-12 mt-2"
            >
              <div
                class="form-outline"
              >
                <label
                  class="d-md-none d-sm-block text-center mb-1"
                >
                  Tc Kimlik Numarası
                </label>
                <input
                  class="w-100 border rounded"
                  name="TCNumber"
                  placeholder="Tc Kimlik Numarası"
                  value=""
                  values=""
                />
              </div>
            </div>
          </div>
          <div
            class="row mb-md-3 mb-sm-1"
          >
            <div
              class="col-md-6 col-sm-12 mt-2"
            >
              <div
                class="form-outline"
              >
                <label
                  class="d-md-none d-sm-block text-center mb-1"
                >
                  İsim Soyisim
                </label>
                <input
                  class="w-100 border rounded"
                  name="fullName"
                  placeholder="İsim Soyisim"
                  value=""
                  values=""
                />
              </div>
            </div>
            <div
              class="col-md-6 col-sm-12 mt-2"
            >
              <div
                class="form-outline"
              >
                <label
                  class="d-md-none d-sm-block text-center mb-1"
                >
                  Satıcı Telefon Numarası
                </label>
                <input
                  class="w-100 border rounded"
                  name="sellerPhoneNumber"
                  placeholder="Satıcı Telefon Numarası"
                  value=""
                  values=""
                />
              </div>
            </div>
          </div>
          <div
            class="row mb-md-3 mb-sm-1"
          >
            <div
              class="col-md-6 col-sm-12 mt-2"
            >
              <div
                class="form-outline"
              >
                <label
                  class="d-md-none d-sm-block text-center mb-1"
                >
                  Satıcı Email
                </label>
                <input
                  class="w-100 border rounded"
                  name="sellerEmail"
                  placeholder="Satıcı Email"
                  value=""
                  values=""
                />
              </div>
            </div>
            <div
              class="col-md-6 col-sm-12 mt-2"
            >
              <div
                class="form-outline"
              >
                <label
                  class="d-md-none d-sm-block text-center mb-1"
                >
                  Alıcı Telefon Numarası
                </label>
                <input
                  class="w-100 border rounded"
                  name="buyerPhoneNumber"
                  placeholder="Alıcı Telefon Numarası"
                  value=""
                  values=""
                />
              </div>
            </div>
          </div>
          <div
            class="row mb-md-3 mb-sm-1"
          >
            <div
              class="col-md-6 col-sm-12 mt-2"
            >
              <div
                class="form-outline"
              >
                <label
                  class="d-md-none d-sm-block text-center mb-1"
                >
                  Alıcı Email
                </label>
                <input
                  class="w-100 border rounded"
                  name="buyerEmail"
                  placeholder="Alıcı Email"
                  value=""
                  values=""
                />
              </div>
            </div>
            <div
              class="col-md-6 col-sm-12 mt-2"
            >
              <div
                class="form-outline"
              >
                <label
                  class="d-md-none d-sm-block text-center mb-1"
                >
                  Satıcı IBAN
                </label>
                <input
                  class="w-100 border rounded"
                  name="sellerIBAN"
                  placeholder="Satıcı IBAN"
                  value=""
                  values=""
                />
              </div>
            </div>
          </div>
          <div
            class="row mb-md-3 mb-sm-1"
          >
            <div
              class="col-md-6 col-sm-12 mt-2"
            >
              <div
                class="form-outline"
              >
                <label
                  class="d-md-none d-sm-block text-center mb-1"
                >
                  Araç Plaka Numarası
                </label>
                <input
                  class="w-100 border rounded"
                  name="vehicleNumber"
                  placeholder="Araç Plaka Numarası"
                  value=""
                  values=""
                />
              </div>
            </div>
            <div
              class="col-md-6 col-sm-12 mt-2"
            >
              <div
                class="form-outline"
              >
                <label
                  class="d-md-none d-sm-block text-center mb-1"
                >
                  Satış Tutarı
                </label>
                <input
                  class="w-100 border rounded"
                  name="price"
                  placeholder="Satış Tutarı"
                  value=""
                  values=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="border rounded px-3 pt-2"
        >
          <label
            class="mb-3 d-block"
          >
            <input
              class="me-1"
              name="consentTextKVKK"
              type="checkbox"
              value="false"
              values="false"
            />
            Tarafıma hizmet sunulması amacıyla, bulut tabanlı yazılımların yurt dışında tutulması nedeniyle kişisel verilerimin yurt dışına aktarılmasına ilişkin Açık Rıza Metni'ni okudum. Kişisel verilerimin bu kapsamda aktarılmasına onay veriyorum.
          </label>
          <label
            class="mb-3 d-block"
          >
            <input
              class="me-1"
              name="consentTextCompany"
              type="checkbox"
              value="false"
              values="false"
            />
            TURK Elektronik Para A.Ş. tarafından; yukarıda yer alan iletişim bilgilerime, bilgilendirme, mal ve hizmet tanıtımı, reklam ve promosyon gibi amaçlarla yararlı olabilecek içeriklere ulaşabilmem için ticari elektronik ileti gönderilmesini her zaman reddetme hakkımın saklı olduğunu bilerek onay veriyorum.
          </label>
          <span
            class="mb-3 d-block"
          >
            Kişisel verileriniz talebinizin gerçekleştirilmesi amacıyla 6698 Sayılı Kanun kapsamında işlenmektedir. Aydınlatma Metnine 
            <b>
              buradan
            </b>
             ulaşabilirsiniz.
          </span>
        </div>
        <div
          class="d-flex justify-content-center mt-2"
        >
          <button
            class="button"
            type="submit"
          >
            <span
              style="color: rgb(221, 221, 221);"
            >
              Gönder
            </span>
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      action="#"
    >
      <div
        class="border rounded px-3 pt-2"
      >
        <div
          class="row mb-md-2 mb-sm-1"
        >
          <div
            class="col-md-6 col-sm-12 mt-2"
          >
            <div
              class="form-outline"
            >
              <select
                class="w-100 border rounded"
                name="legalOrRealPerson"
                values="gerçek"
              >
                <option
                  value="gerçek"
                >
                  Gerçek Kişi
                </option>
                <option
                  value="tüzel"
                >
                  Tüzel Kişi
                </option>
              </select>
            </div>
          </div>
          <div
            class="col-md-6 col-sm-12 mt-2"
          >
            <div
              class="form-outline"
            >
              <label
                class="d-md-none d-sm-block text-center mb-1"
              >
                Tc Kimlik Numarası
              </label>
              <input
                class="w-100 border rounded"
                name="TCNumber"
                placeholder="Tc Kimlik Numarası"
                value=""
                values=""
              />
            </div>
          </div>
        </div>
        <div
          class="row mb-md-3 mb-sm-1"
        >
          <div
            class="col-md-6 col-sm-12 mt-2"
          >
            <div
              class="form-outline"
            >
              <label
                class="d-md-none d-sm-block text-center mb-1"
              >
                İsim Soyisim
              </label>
              <input
                class="w-100 border rounded"
                name="fullName"
                placeholder="İsim Soyisim"
                value=""
                values=""
              />
            </div>
          </div>
          <div
            class="col-md-6 col-sm-12 mt-2"
          >
            <div
              class="form-outline"
            >
              <label
                class="d-md-none d-sm-block text-center mb-1"
              >
                Satıcı Telefon Numarası
              </label>
              <input
                class="w-100 border rounded"
                name="sellerPhoneNumber"
                placeholder="Satıcı Telefon Numarası"
                value=""
                values=""
              />
            </div>
          </div>
        </div>
        <div
          class="row mb-md-3 mb-sm-1"
        >
          <div
            class="col-md-6 col-sm-12 mt-2"
          >
            <div
              class="form-outline"
            >
              <label
                class="d-md-none d-sm-block text-center mb-1"
              >
                Satıcı Email
              </label>
              <input
                class="w-100 border rounded"
                name="sellerEmail"
                placeholder="Satıcı Email"
                value=""
                values=""
              />
            </div>
          </div>
          <div
            class="col-md-6 col-sm-12 mt-2"
          >
            <div
              class="form-outline"
            >
              <label
                class="d-md-none d-sm-block text-center mb-1"
              >
                Alıcı Telefon Numarası
              </label>
              <input
                class="w-100 border rounded"
                name="buyerPhoneNumber"
                placeholder="Alıcı Telefon Numarası"
                value=""
                values=""
              />
            </div>
          </div>
        </div>
        <div
          class="row mb-md-3 mb-sm-1"
        >
          <div
            class="col-md-6 col-sm-12 mt-2"
          >
            <div
              class="form-outline"
            >
              <label
                class="d-md-none d-sm-block text-center mb-1"
              >
                Alıcı Email
              </label>
              <input
                class="w-100 border rounded"
                name="buyerEmail"
                placeholder="Alıcı Email"
                value=""
                values=""
              />
            </div>
          </div>
          <div
            class="col-md-6 col-sm-12 mt-2"
          >
            <div
              class="form-outline"
            >
              <label
                class="d-md-none d-sm-block text-center mb-1"
              >
                Satıcı IBAN
              </label>
              <input
                class="w-100 border rounded"
                name="sellerIBAN"
                placeholder="Satıcı IBAN"
                value=""
                values=""
              />
            </div>
          </div>
        </div>
        <div
          class="row mb-md-3 mb-sm-1"
        >
          <div
            class="col-md-6 col-sm-12 mt-2"
          >
            <div
              class="form-outline"
            >
              <label
                class="d-md-none d-sm-block text-center mb-1"
              >
                Araç Plaka Numarası
              </label>
              <input
                class="w-100 border rounded"
                name="vehicleNumber"
                placeholder="Araç Plaka Numarası"
                value=""
                values=""
              />
            </div>
          </div>
          <div
            class="col-md-6 col-sm-12 mt-2"
          >
            <div
              class="form-outline"
            >
              <label
                class="d-md-none d-sm-block text-center mb-1"
              >
                Satış Tutarı
              </label>
              <input
                class="w-100 border rounded"
                name="price"
                placeholder="Satış Tutarı"
                value=""
                values=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="border rounded px-3 pt-2"
      >
        <label
          class="mb-3 d-block"
        >
          <input
            class="me-1"
            name="consentTextKVKK"
            type="checkbox"
            value="false"
            values="false"
          />
          Tarafıma hizmet sunulması amacıyla, bulut tabanlı yazılımların yurt dışında tutulması nedeniyle kişisel verilerimin yurt dışına aktarılmasına ilişkin Açık Rıza Metni'ni okudum. Kişisel verilerimin bu kapsamda aktarılmasına onay veriyorum.
        </label>
        <label
          class="mb-3 d-block"
        >
          <input
            class="me-1"
            name="consentTextCompany"
            type="checkbox"
            value="false"
            values="false"
          />
          TURK Elektronik Para A.Ş. tarafından; yukarıda yer alan iletişim bilgilerime, bilgilendirme, mal ve hizmet tanıtımı, reklam ve promosyon gibi amaçlarla yararlı olabilecek içeriklere ulaşabilmem için ticari elektronik ileti gönderilmesini her zaman reddetme hakkımın saklı olduğunu bilerek onay veriyorum.
        </label>
        <span
          class="mb-3 d-block"
        >
          Kişisel verileriniz talebinizin gerçekleştirilmesi amacıyla 6698 Sayılı Kanun kapsamında işlenmektedir. Aydınlatma Metnine 
          <b>
            buradan
          </b>
           ulaşabilirsiniz.
        </span>
      </div>
      <div
        class="d-flex justify-content-center mt-2"
      >
        <button
          class="button"
          type="submit"
        >
          <span
            style="color: rgb(221, 221, 221);"
          >
            Gönder
          </span>
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
